src/Helper.s

```diff
@@ -156,19 +194,23 @@ GETBLOCK:   PHX               ; save X
 ;
 ;*******************************
 
-COMMAND:    PHY               ; save Y
+COMMAND:    
+            JSR   WREADY
+            BCC   @CONT
+            RTS
+@CONT:      PHY               ; save Y
             LDY   SLOT
-            STA   DATA,X      ; send command
+            JSR   WRDATA      ; send command
```

Comment:
This adds an overhead of several cycles and may not be needed, imho.

Reply:
Your are right for 7MHz clock, but I am sending commands at low frequency clock for reasons I explain in comment below. For 7Mhz these are not needed.